includes/filename.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+// Exit if accessed directly
+if(!defined('ABSPATH')){
+	exit;
+}
+
+/**
+ * Default functions
+ */
+class Fiber_Admin_Filename{
+	public function __construct(){
+		// Auto Convert File Name
+		if(fiad_get_miscellaneous_option('auto_convert_file_name')){
+			add_filter('sanitize_file_name', [$this, 'fiad_auto_convert_file_name'], 10, 2);
+		}
+	}
+	
+	// Auto Convert File Name
+	public function fiad_auto_convert_file_name($filename, $filename_raw){
+		//handle urlencode case
```

Comment:
Could you give an example of the file name before and after this code? I'm not sure how you handle the urlencode here.

Reply:
@daomapsieucap I have added the following example to comment in the code also

```
// Before: Really%20Ugly%20Filename--That-Is_Too Common…..png
// Default of WordPress: ReAlly20Ugly20Filename-_-That_-_Is_Too-Common….pdf (still remains the '20')
// After ReAllyUglyFilename-_-That_-_Is_Too-Common….pdf (still remains some special char but get rid of the '20')
```